connector/src/main/java/org/jboss/as/connector/services/datasources/statistics/DataSourceStatisticsService.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2011, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.connector.services.datasources.statistics;
+
+import static org.jboss.as.connector.logging.ConnectorLogger.ROOT_LOGGER;
+
+import org.jboss.as.connector.dynamicresource.StatisticsResourceDefinition;
+import org.jboss.as.connector.subsystems.datasources.DataSourcesSubsystemProviders;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.registry.PlaceholderResource;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.jca.core.spi.statistics.StatisticsPlugin;
+import org.jboss.jca.deployers.common.CommonDeployment;
+import org.jboss.msc.inject.Injector;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StartException;
+import org.jboss.msc.service.StopContext;
+import org.jboss.msc.value.InjectedValue;
+
+public class DataSourceStatisticsService implements Service<ManagementResourceRegistration> {
```

Comment:
I marked this Ready for Merge because I don't want to block over this, but I don't see why this isn't Service<Void>. I don't want to expose ManagementResourceRegistration via MSC unless it's critical to do so.